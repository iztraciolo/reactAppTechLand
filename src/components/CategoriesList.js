import React, {useEffect, useState} from "react";


function CategoriesList() {

  const [categories, setCategories] = useState([]);

  const apiProducts = `/api/products?page=1`;



useEffect(()=>{
    console.log("DidMount");
     const fetchData = (url)=>{
        fetch(url)
        .then(response=>response.json())
        .then(data=>{
            console.log(data);
            setCategories(data.countByCategory);
        })
        .catch(error=>console.log(error));            
     }

     fetchData(apiProducts);
    
}, []);

let arrayData = [];
for (const [key, value] of Object.entries(categories)) {
  arrayData.push(`${key}: ${value}`);
}


  
  return (
    <div className="col-lg-6 mb-4">
      <div className="card shadow mb-4">
        <div className="card-header py-3">
          <h5 className="m-0 font-weight-bold text-gray-800">
            Categor√≠as 
          </h5>
        </div>
        <div className="card-body">
          <div className="row">
            
          { 
          
          arrayData.map((cate, i)=>{
            return(
              <div className="col-lg-6 mb-4">
                  <div className="card bg-dark text-white shadow">
                    <div className="card-body">{cate}</div>
                  </div>
              </div>
          )
            
          })

         

         

            
            
          }

         
            
            
            


            
          </div>
        </div>
      </div>
    </div>
  );
}

export default CategoriesList;
