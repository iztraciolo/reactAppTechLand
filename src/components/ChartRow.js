import React from 'react';


function ChartRow(props){
    return (
                <tr>
                    <td>{props.name_product}</td>
                    <td>
                        <img src={props.image_product} style={{width: 10 +'rem'}}  />                        
                    </td>
                    <td>{props.description}</td>
                    <td>{props.brands.name_brand}</td>
                    
                    <td>
                        <ul>
                            {props.categories.map( (category,i) => 
                                <li key={`category ${i}`}>{category.name_category}</li>
                            )}
                        </ul>
                    </td>
                    <td>
                        <ul>
                            {props.typeComponent.map( (typeCompo,i) => 
                                <li key={`typeCompo ${i}`}>{typeCompo.name_type_component}</li>
                            )}
                        </ul>
                    </td>
                   
                    
                </tr>
            )
    }
    
        

export default ChartRow;