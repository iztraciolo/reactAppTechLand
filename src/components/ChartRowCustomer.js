import React from 'react';
import { Link } from 'react-router-dom';


function ChartRowCustomer(props){
    return (
                <tr>
                    <td>
                        <Link to={props.detail}>{props.customer_id}</Link>                        
                    </td>
                    
                    <td>
                        <Link to={props.detail}>{props.full_name}</Link>                        
                    </td>

                    <td>
                        <Link to={props.detail}>{props.email}</Link>                        
                    </td>
                    <td>
                        <Link to={props.detail}>
                            <img src={props.customer_profile_image} style={{width: 10 +'rem'}}  />
                        </Link>                                                    
                    </td>
                </tr>
            )
    }
    
        

export default ChartRowCustomer;