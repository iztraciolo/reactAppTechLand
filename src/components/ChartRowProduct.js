import React from 'react';
import { Link } from 'react-router-dom';


function ChartRowProduct(props){
    return (
                <tr>
                    <td>
                        <Link to={props.detail}>{props.name_product}</Link>                        
                    </td>

                    <td>
                        <img src={props.image_product} style={{width: 10 +'rem'}}  />                        
                    </td>
                    <td>{props.description}</td>
                    <td>{props.brands.name_brand}</td>
                    
                    <td>
                        <ul>
                            {props.categories.map( (category,i) => 
                                <li key={`category ${i}`}>{category.name_category}</li>
                            )}
                        </ul>
                    </td>
                    <td>
                        <ul>
                            {props.typeComponent.map( (typeCompo,i) => 
                                <li key={`typeCompo ${i}`}>{typeCompo.name_type_component}</li>
                            )}
                        </ul>
                    </td>
                   
                    
                </tr>
            )
    }
    
        

export default ChartRowProduct;