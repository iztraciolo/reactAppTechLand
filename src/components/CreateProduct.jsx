import TopBar from "./TopBar";
import Footer from "./Footer";
import React, {useState} from 'react';


/** @Author: Erick Luna
 ** Componente React que permite registrar un producto utilizando fetch y el método post
 * */

/*Los arreglos con información para las opciones de las etiquetas select se importaron a otro archivo,
 para mayor legibilidad del código
*/
import {types, brands, categories} from "../model/data";

export async function fetchRegister(url, data) {
    return await fetch(url, {
        method: 'POST',
        headers: {
            'Accept': 'application/json',
            'Content-Type': 'application/json',
        },
        body: JSON.stringify({
            ...data
        })
    });
}


const url = "http://localhost:3001/products/create";


/*Se migro este formulario de el sitio "Tech Land a un "Dashboard", pues la funcionalidad de "Añadir producto"
debe estar limitada a los administradores del sistema*/

export function CreateProduct(){
    
    const [name, setName] = useState();
    const [description, setDescription] = useState();
    const [category, setCategory] = useState();
    const [brand, setBrand] = useState();
    const [type, setType]=useState();
    const [price, setPrice] = useState();
    const [stock, setStock] = useState();
    
    function changeName(event){
        setName(event.target.value);
        console.log(name)
    }
    
    function changeDescription(event){
        setDescription(event.target.value);
    }
    
    function changeCategory(event){
        setCategory(event.target.value);
    }
    function changeBrand(event){
        setBrand(event.target.value);
    }
    
    function changeType(event){
        setType(event.target.value);
    }
    
    function changePrice(event){
        setPrice(event.target.value);
    }
    
    function changeStock(event){
        setStock(event.target.value);
    }
    
    /** Esta función se ejecuta al presionar el botón de enviar; permite realizar
     **  la petición por "POST" a nuestra Api rest
     */
    
    
    function handleSubmit(event) {
        event.preventDefault();
        fetch(url,{
            method:'POST',
            headers:{
                'Accept': 'application/json',
                'Content-Type': 'application/json',
            },
            body: JSON.stringify({
                name_product: name,
                description: description,
                image_product: "",
                category: category,
                brand: brand,
                price:price,
                stock: stock,
                type: type
            })
        }).then(response=>{
            console.log(response);
        })
    }
    
    
    //Este return renderiza el contenido; el html es mas limpio gracias al uso del map para generar los <options>
    return(
        <React.Fragment>
            {/*<!-- Content Wrapper -->*/}
            <div id="content-wrapper" className="d-flex flex-column">
                {/*<!-- Main Content -->*/}
                <div id="content">
                    <TopBar />
                    <div className="badge-dark mt-5 container-fluid w-50 mx-auto p-4 rounded">
                        <h3 className="text-center mx-auto alert-secondary rounded p-2 mb-2 w-50">
                            Añadir producto
                        </h3>
                        <br/>
                        <form className="mb-3">
                            <div className="form-group">
                                <label className="form-check-label">Nombre del producto </label>
                                <input onChange={changeName} type="text" className="input-group"/>
                            </div>
                            <div className="form-group">
                                <label className="form-check-label">Descripción</label>
                                <textarea onChange={changeDescription} className="input-group"/>
                            </div>
                            <div className="form-group">
                                <label className="form-check-label">Imagen </label>
                                <input  type="file" className="input-group"/>
                            </div>
                            <div className="form-group">
                                <label className="form-check-label">Categoría </label>
                                <select onChange={changeCategory} className="input-group" id="category">
                                    {/* Se utiliza un ciclo map, que permite recorrer un arreglo y retorna la etiqueta
                                    <option> con los valores requeridos*/}
                                    {categories.map((cat, i)=>{
                                        return <option key={i+cat} value={cat}>{cat}</option>
                                    })}
                                </select>
                            </div>
                            <div className="form-group">
                                <label className="form-check-label">Marca </label>
                                <select onChange={changeBrand} className="input-group" id="brand">
                                    {brands.map((brand, i)=>{
                                        return <option key={i+brand} value={brand}>{brand}</option>
                                    })}
                                </select>
                            </div>
                            <div className="form-group">
                                <label className="form-check-label">Precio </label>
                                <input onChange={changePrice} type="number" className="input-group"/>
                            </div>
                            <div className="form-group">
                                <label className="form-check-label">Stock </label>
                                <input type="text" onChange={changeStock} className="input-group"/>
                            </div>
                            <div className="form-group">
                                <label className="form-check-label">Tipo </label>
                                <select onChange={changeType} className="input-group" id="type">
                                    {types.map((ty, i)=>{
                                        return <option key={i+ty} value={ty}>{ty}</option>
                                    })}
                                </select>
                            </div>
                            <div className="form-group">
                            <button onClick={handleSubmit} type="submit" className="btn-light w-50 d-block mx-auto rounded"> Agregar</button>
                            </div>
                        </form>
                    
                    </div>
                    <Footer />
                </div>
            </div>
        </React.Fragment>
    );
}