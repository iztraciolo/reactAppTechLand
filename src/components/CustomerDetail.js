import {useState, useEffect} from 'react';

function CustomerDetail(props){


    const id = props.match.params.id;
    const apiCustomerDetail = `/api/customer/${id}`;

    const [customer, setCustomer] = useState([]);

   

    useEffect(()=>{
        console.log("DidMountCustomerDetail");
        const fetchData = (url)=>{
            fetch(url)
            .then(response=>response.json())
            .then(data=>{
                console.log(data);      
                setCustomer(data);                
            })
            .catch(error=>console.log(error));            
     }

     fetchData(apiCustomerDetail);
    }, [])

    


    return(
        <div className="col-lg-6 mb-4">
            <div className="card shadow mb-4">
                <div className="card-header py-3">
                    <h5 className="m-0 font-weight-bold text-gray-800">Cliente {customer.full_name}</h5>
                </div>
                <div className="card-body">
                    <div className="text-center">
                        <img className="img-fluid px-3 px-sm-4 mt-3 mb-4" style={{width: 40 +'rem'}} src={customer.customer_profile_image} alt=" perfil del cliente "/>
                    </div>
                    
                    <dl className="row">
                                <dt className="col-sm-3">Nombre del cliente</dt>
                                <dd className="col-sm-9">{customer.full_name}</dd>

                                <dt className="col-sm-3">Email</dt>
                                <dd className="col-sm-9">{customer.email}</dd>

                                <dt className="col-sm-3">Tel√©fono</dt>
                                <dd className="col-sm-9">{customer.customer_number_phone}</dd>

                                <dt className="col-sm-3">Nacionalidad</dt>
                                <dd className="col-sm-9">{customer.nationality}</dd>

                                

                               
                            </dl>
                    
                </div>
            </div>
        </div>
    )
}

export default CustomerDetail;
