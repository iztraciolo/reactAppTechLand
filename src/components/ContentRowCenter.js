import React from 'react';
import CategoriesList from './CategoriesList';
import LastProduct from './LastProduct';

function ContentRowCenter(){
    return (
        <div className="row">
            
            {/*<!-- Last Movie in DB -->*/}
            <LastProduct />
            {/*<!-- End content row last movie in Data Base -->*/}

            {/*<!-- Genres in DB -->*/}
            <CategoriesList />

        </div>
    )
}

export default ContentRowCenter;