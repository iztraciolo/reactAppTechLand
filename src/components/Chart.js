import {useState, useEffect} from 'react';
import ChartRow from './ChartRow';

let tableRowsData = [
    {
        Title: 'Billy Elliot ',
        Length: '123',
        Rating: '5',
        Categories: ['Drama','Comedia'],
        Awards: 2
    },
    {
        Title: 'Alicia en el país de las maravillas',
        Length: '142',
        Rating: '4.8',
        Categories: ['Drama','Acción','Comedia'],
        Awards: 3
    },
    
]


function Chart (){
    // calling the api to get products info

const [products, setProducts] = useState([]);
const [page, setPage] = useState(1);


useEffect(()=>{
    console.log("DidMount");
     const fetchData = async (url)=>{
       await fetch(url)
        .then(response=>response.json())
        .then(data=>{
            console.log(data);
            setProducts(data);
        })            
     }

     fetchData(`/api/products?page=${page}`)
    
}, [])

    


    return (
        /* <!-- DataTales Example --> */
        <div className="card shadow mb-4">
            <div className="card-body">
                <div className="table-responsive">
                    <table className="table table-bordered" id="dataTable" width="100%" cellSpacing="0">
                        <thead>
                            <tr>
                                <th>Título</th>
                                <th>Duración</th>
                                <th>Rating</th>
                                <th>Género</th>
                                <th>Premios</th>
                            </tr>
                        </thead>
                        <tfoot>
                            <tr>
                                <th>Título</th>
                                <th>Duración</th>
                                <th>Rating</th>
                                <th>Género</th>
                                <th>Premios</th>
                            </tr>
                        </tfoot>
                        <tbody>
                            {
                            tableRowsData.map( ( row , i) => {
                                return <ChartRow { ...row} key={i}/>
                            })
                            }

                        </tbody>
                    </table>
                </div>
            </div>
        </div>

    )
}

export default Chart;