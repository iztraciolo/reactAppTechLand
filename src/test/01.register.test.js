import { shallow } from 'enzyme';
import {CreateProduct} from "../components/CreateProduct";

describe("Pruebas de renderizado", ()=>{
    
    test("Debe renderizar correctamente <CreateProduct/>",()=>{
       const wrapper = shallow(<CreateProduct/>);
       
       expect(wrapper).toMatchSnapshot();
    });
    
    
});