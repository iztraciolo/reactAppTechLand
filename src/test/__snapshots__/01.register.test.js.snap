// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de renderizado Debe renderizar correctamente <CreateProduct/> 1`] = `
<Fragment>
  <div
    className="d-flex flex-column"
    id="content-wrapper"
  >
    <div
      id="content"
    >
      <TopBar />
      <div
        className="badge-dark mt-5 container-fluid w-50 mx-auto p-4 rounded"
      >
        <h3
          className="text-center mx-auto alert-secondary rounded p-2 mb-2 w-50"
        >
          Añadir producto
        </h3>
        <br />
        <form
          className="mb-3"
        >
          <div
            className="form-group"
          >
            <label
              className="form-check-label"
            >
              Nombre del producto 
            </label>
            <input
              className="input-group"
              onChange={[Function]}
              type="text"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-check-label"
            >
              Descripción
            </label>
            <textarea
              className="input-group"
              onChange={[Function]}
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-check-label"
            >
              Imagen 
            </label>
            <input
              className="input-group"
              type="file"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-check-label"
            >
              Categoría 
            </label>
            <select
              className="input-group"
              id="category"
              onChange={[Function]}
            >
              <option
                key="0Perifericos"
                value="Perifericos"
              >
                Perifericos
              </option>
              <option
                key="1Equipos de computo"
                value="Equipos de computo"
              >
                Equipos de computo
              </option>
              <option
                key="2Laptops"
                value="Laptops"
              >
                Laptops
              </option>
              <option
                key="3Monitores"
                value="Monitores"
              >
                Monitores
              </option>
              <option
                key="4Procesadores"
                value="Procesadores"
              >
                Procesadores
              </option>
              <option
                key="5Tarjetas Madre"
                value="Tarjetas Madre"
              >
                Tarjetas Madre
              </option>
              <option
                key="6Tarjetas de video"
                value="Tarjetas de video"
              >
                Tarjetas de video
              </option>
              <option
                key="7 Disipadores y Enfriamiento"
                value=" Disipadores y Enfriamiento"
              >
                 Disipadores y Enfriamiento
              </option>
              <option
                key="8PC Gamer"
                value="PC Gamer"
              >
                PC Gamer
              </option>
              <option
                key="9Almacenamiento"
                value="Almacenamiento"
              >
                Almacenamiento
              </option>
              <option
                key="10Memorias"
                value="Memorias"
              >
                Memorias
              </option>
              <option
                key="11Otra"
                value="Otra"
              >
                Otra
              </option>
            </select>
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-check-label"
            >
              Marca 
            </label>
            <select
              className="input-group"
              id="brand"
              onChange={[Function]}
            >
              <option
                key="0Acer"
                value="Acer"
              >
                Acer
              </option>
              <option
                key="1Adata"
                value="Adata"
              >
                Adata
              </option>
              <option
                key="2AlienWare"
                value="AlienWare"
              >
                AlienWare
              </option>
              <option
                key="3AMD"
                value="AMD"
              >
                AMD
              </option>
              <option
                key="4Aorus"
                value="Aorus"
              >
                Aorus
              </option>
              <option
                key="5Apple"
                value="Apple"
              >
                Apple
              </option>
              <option
                key="6Asus"
                value="Asus"
              >
                Asus
              </option>
              <option
                key="7Dell"
                value="Dell"
              >
                Dell
              </option>
              <option
                key="8EVGA"
                value="EVGA"
              >
                EVGA
              </option>
              <option
                key="9Gigabyte"
                value="Gigabyte"
              >
                Gigabyte
              </option>
              <option
                key="10HP"
                value="HP"
              >
                HP
              </option>
              <option
                key="11Huawei"
                value="Huawei"
              >
                Huawei
              </option>
              <option
                key="12HyperX"
                value="HyperX"
              >
                HyperX
              </option>
              <option
                key="13Intel"
                value="Intel"
              >
                Intel
              </option>
              <option
                key="14Kingston"
                value="Kingston"
              >
                Kingston
              </option>
              <option
                key="15Lenovo"
                value="Lenovo"
              >
                Lenovo
              </option>
              <option
                key="16MSI"
                value="MSI"
              >
                MSI
              </option>
              <option
                key="17Nvidia"
                value="Nvidia"
              >
                Nvidia
              </option>
              <option
                key="18Samsung"
                value="Samsung"
              >
                Samsung
              </option>
              <option
                key="19Sony"
                value="Sony"
              >
                Sony
              </option>
              <option
                key="20Otro"
                value="Otro"
              >
                Otro
              </option>
            </select>
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-check-label"
            >
              Precio 
            </label>
            <input
              className="input-group"
              onChange={[Function]}
              type="number"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-check-label"
            >
              Stock 
            </label>
            <input
              className="input-group"
              onChange={[Function]}
              type="text"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-check-label"
            >
              Tipo 
            </label>
            <select
              className="input-group"
              id="type"
              onChange={[Function]}
            >
              <option
                key="0Destacados"
                value="Destacados"
              >
                Destacados
              </option>
              <option
                key="1Agregados recientemente"
                value="Agregados recientemente"
              >
                Agregados recientemente
              </option>
              <option
                key="2Más vendidos"
                value="Más vendidos"
              >
                Más vendidos
              </option>
              <option
                key="3En oferta"
                value="En oferta"
              >
                En oferta
              </option>
            </select>
          </div>
          <div
            className="form-group"
          >
            <button
              className="btn-light w-50 d-block mx-auto rounded"
              onClick={[Function]}
              type="submit"
            >
               Agregar
            </button>
          </div>
        </form>
      </div>
      <Footer />
    </div>
  </div>
</Fragment>
`;
